src/test/regress/expected/multi_subquery_complex_reference_clause.out

```diff
@@ -150,30 +150,46 @@ WHERE subquery_1.user_id = subquery_2.user_id ;
      4
 (1 row)
 
-  -- Should be able to push down since reference tables are inner joined
-  -- with hash distributed tables, the results of those joins are the parts of
-  -- an outer join
-SELECT subquery_2.id FROM
-  (SELECT tt1.user_id, random() FROM user_buy_test_table AS tt1 JOIN users_return_test_table as tt2
-    ON tt1.user_id = tt2.user_id) subquery_1
-  RIGHT JOIN
-  (SELECT tt1.user_id, ref.id, random() FROM user_buy_test_table as tt1 JOIN users_ref_test_table as ref
-   ON tt1.user_id = ref.id) subquery_2 ON subquery_1.user_id = subquery_2.user_id ORDER BY 1 DESC LIMIT 5;
+-- Should be able to push down since reference tables are inner joined
```

Comment:
As far as I can see, we don't have tests that triggers the following two cases:

- We're recursively planning an unrelated part of the query, but still be able to pushdown the UNION query
- We're recursively planning an unrelated part of the query, and cannot pushdown the UNION because there is no distribution key equality.

I can hear your complaints that these cases are not directly related to your changes. That's true, they are semi-related since they involve reference table JOIN dist table, and unrelated intermediate results. Still, I'd be happy to have these tests as well.

I'm giving two example queries based on the tables [here](https://github.com/citusdata/citus/issues/2518#issue-387260734). 


```SQL

SELECT *
FROM
  (SELECT test.x,
          test.y
   FROM REF
   JOIN test ON (test.x = ref.a)
   WHERE a IN
       (SELECT y
        FROM test)
   UNION SELECT x,
                y
   FROM test
   WHERE x IN
       (SELECT a
        FROM REF) ) AS fpp;

        
SELECT *
FROM
  (SELECT test.x,
          test.y
   FROM REF
   JOIN test ON (test.x = ref.a)
   WHERE a IN
       (SELECT y
        FROM test)
   UNION SELECT y,
                x
   FROM test
   WHERE x IN
       (SELECT a
        FROM REF) ) AS fpp;
```